### DISCUSSION TITLE
Custom LSP commands

### DISCUSSION THREAD

**Discussion by dariooddenino** [helix-editor/helix#3933] [Q&A] [answered]
Is there a way to send custom LSP commands to a running server?

**Comment by the-mikedavis** [comment, marked as answer]
Unless you mean code actions (Space-a), this is not possible. There are a few language servers that we know use custom extensions on the LSP protocol (rust-analyzer, solargraphs) and supporting their custom extensions would be great as plugins. Until there's a plugin system, a fork is the most pragmatic way to implement the extensions.

**Comment by dariooddenino** [comment]
Yeah, I need to to send a custom `workspace/executeCommand` (`purescript.build`).

I guess I'll go with a fork for now, thanks!
